diff --git a/node_modules/firebase-admin/lib/utils/api-request.js b/node_modules/firebase-admin/lib/utils/api-request.js
index 37584e2..dbd65bd 100644
--- a/node_modules/firebase-admin/lib/utils/api-request.js
+++ b/node_modules/firebase-admin/lib/utils/api-request.js
@@ -707,7 +707,7 @@ class BaseRequestConfigImpl {
     buildUrl() {
         const fullUrl = this.urlWithProtocol();
         if (!this.hasEntity() || this.isEntityEnclosingRequest()) {
-            return url.parse(fullUrl);
+            return new url.URL(fullUrl);
         }
         if (!validator.isObject(this.data)) {
             throw new Error(`${this.method} requests cannot have a body`);
@@ -720,7 +720,7 @@ class BaseRequestConfigImpl {
                 parsedUrl.searchParams.append(key, dataObj[key]);
             }
         }
-        return url.parse(parsedUrl.toString());
+        return parsedUrl;
     }
     urlWithProtocol() {
         const fullUrl = this.url;
@@ -760,7 +760,7 @@ class HttpRequestConfigImpl extends BaseRequestConfigImpl {
             protocol,
             hostname: parsed.hostname,
             port,
-            path: parsed.path,
+            path: parsed.pathname + (parsed.search || ''),
             method: this.method,
             agent: this.httpAgent,
             headers: Object.assign({}, this.headers),
@@ -784,7 +784,7 @@ class Http2RequestConfigImpl extends BaseRequestConfigImpl {
         const protocol = parsed.protocol;
         return {
             protocol,
-            path: parsed.path,
+            path: parsed.pathname + (parsed.search || ''),
             method: this.method,
             headers: Object.assign({}, this.headers),
         };
